src/node_state.rs

```diff
@@ -0,0 +1,343 @@
+        let dropped = self.timeouts.drop(height);
+        for tx in dropped {
+            self.unsettled_transactions.remove(&tx);
+        }
+    }
+
+    fn update_state_tx(
+        &mut self,
+        tx: &ProofTransaction,
+        blob_detail: UnsettledBlobDetail,
+    ) -> Result<(), Error> {
+        let unsettled_tx = match self.unsettled_transactions.get_mut(&tx.tx_hash) {
+            Some(tx) => tx,
+            None => bail!("Tx is either settled or does not exists."),
+        };
+
+        // TODO: better not using "as usize"
+        unsettled_tx.blobs[tx.blob_index.0 as usize] = blob_detail;
+
+        Ok(())
+    }
+
+    fn is_settled(&self, tx_hash: &TxHash) -> bool {
+        let tx = match self.unsettled_transactions.get(tx_hash) {
+            Some(tx) => tx,
+            None => {
+                return false;
+            }
+        };
+
+        tx.blobs
+            .iter()
+            .all(|blob| blob.verification_status == VerificationStatus::Success)
+    }
+
+    fn verify_proof(tx: &ProofTransaction) -> Result<UnsettledBlobDetail, Error> {
+        // TODO real implementation
+        Ok(UnsettledBlobDetail {
+            contract_name: tx.contract_name.clone(),
+            verification_status: VerificationStatus::Success,
+            hyle_output: Some(model::HyleOutput {
+                version: 1,
+                initial_state: vec![0, 1, 2, 3],
+                next_state: vec![4, 5, 6],
+                identity: Identity("test".to_string()),
+                tx_hash: tx.tx_hash.clone(),
+                index: tx.blob_index.clone(),
+                blobs: vec![],
+                success: true,
+            }),
+        })
+    }
+
+    fn extract_blobs_hash(blob: &UnsettledBlobDetail) -> Result<BlobsHash, Error> {
+        if let Some(_output) = &blob.hyle_output {
+            Ok(BlobsHash::new("111"))
+        } else {
+            bail!("Missing output in blob details!");
+        }
+    }
+
+    fn settle_tx(&mut self, tx: &ProofTransaction) -> Result<(), Error> {
+        info!("Settle tx {:?}", tx);
+        let unsettled_tx = match self.unsettled_transactions.get_mut(&tx.tx_hash) {
+            Some(tx) => tx,
+            None => bail!("Tx to settle not found!"),
+        };
+        let contracts = unsettled_tx
+            .blobs
+            .iter()
+            .map(|b| b.contract_name.clone())
+            .collect::<Vec<ContractName>>();
+
+        for contract_name in &contracts {
+            self.update_state_contract(contract_name, &tx.tx_hash)?;
+        }
+
+        Ok(())
+    }
+
+    fn update_state_contract(
+        &mut self,
+        contract_name: &ContractName,
+        tx: &TxHash,
+    ) -> Result<(), Error> {
+        let unsettled_tx = match self.unsettled_transactions.get_mut(tx) {
+            Some(tx) => tx,
+            None => bail!("Tx to settle not found!"),
+        };
+        if let Some(contract) = self.contracts.get_mut(contract_name) {
```

Comment:
Style: Pour le coup je trouverais ça plus facile à lire avec l'api `context("not present")`
`If let Ok(contract) = opt.context()? {}`

Reply:
oh oui sûrement, j'ai juste pas du tout le réflexe de anyhow ^^ je vais m'y faire ! 
je corrige ça